BetaCameras/OrbbecOpenNI/OrbbecOpenNI.cpp

```diff
@@ -324,29 +325,26 @@ String^ MetriCam2::Cameras::AstraOpenNI::GetIRFlooderStatus()
 	return statusString;
 }
 
-void MetriCam2::Cameras::AstraOpenNI::SetIRGain(char valueChar)
+void MetriCam2::Cameras::AstraOpenNI::SetIRGain(int value)
 {
-	const char* value;
-	if (valueChar < IR_Gain_MIN)
+	if (value < IR_Gain_MIN)
 	{
-		value = "0x08";
+		value = IR_Gain_MIN;
 	}
-	else if (valueChar > IR_Gain_MAX)
+	else if (value > IR_Gain_MAX)
 	{
-		value = "0x60";
+		value = IR_Gain_MAX;
 	}
-	else
-	{	
-		value = (char*)Marshal::StringToHGlobalAnsi("0x" + Convert::ToString((int)valueChar, 16)).ToPointer();
-	}	
 
-	if (!_pCamData->openNICam->ir_gain_set(value))
+	const char* valueStr = (char*)Marshal::StringToHGlobalAnsi("0x" + Convert::ToString((int)value, 16)).ToPointer();
```

Comment:
Remarks on StringToHGlobalAnsi say: "StringToHGlobalAnsi is useful for custom marshaling or when mixing managed and unmanaged code. Because this method allocates the unmanaged memory required for a string, always free the memory by calling ***FreeHGlobal***. StringToHGlobalAnsi provides the opposite functionality of Marshal.PtrToStringAnsi."